### ISSUE TITLE
portuges.def: error outputing numbers <100 with two strings

### ISSUE THREAD

**Issue by ederguidsa** [Xerdi/fmtcount#8]
In portuguese, either european or brazilian, the numbers below 100 with two strings have an "e" between the strings.

For example: \numberstringnum{55} outputs "cinqüenta cinco" and the correct output is "cinquenta **e** cinco".
This error does not happen for numbers > 100, as far as I have tested. For example \numberstringnum{155} outputs "cento e cinqüenta **e** cinco" as expected.

**Comment by MacLotsen** [comment]
So if I run the `shortbab` tests with portuges I notice that it only counts for `1`, like in `21`. See for example:
```
00021 00015 21o vinte e um
00022 00016 22o vinte dois
00023 00017 23o vinte três
00024 00018 24o vinte quatro
00025 00019 25o vinte cinco
```
Is that incorrect as well, or is this behavior for beyond the 20s? I can confirm that this behavior is also happening for the 50s:
```
00050 00032 50o cinqüenta
00051 00033 51o cinqüenta e um
00055 00037 55o cinqüenta cinco
```
I've got a patch ready which results in:
```
 decimal hexadecimal ordinal numberstring
00000 00000 0 zero
00001 00001 1o um
00002 00002 2o dois
00003 00003 3o três
00004 00004 4o quatro
00005 00005 5o cinco
00006 00006 6o seis
00007 00007 7o sete
00008 00008 8o oito
00009 00009 9o nove
00010 0000a 10o dez
00011 0000b 11o onze
00012 0000c 12o doze
00013 0000d 13o treze
00014 0000e 14o catorze
00015 0000f 15o quinze
00016 00010 16o dezasseis
00017 00011 17o dezassete
00018 00012 18o dezoito
00019 00013 19o dezanove
00020 00014 20o vinte
00021 00015 21o vinte e um
00022 00016 22o vinte e dois
00023 00017 23o vinte e três
00024 00018 24o vinte e quatro
00025 00019 25o vinte e cinco
00030 0001e 30o trinta
00040 00028 40o quarenta
00050 00032 50o cinqüenta
00051 00033 51o cinqüenta e um
00055 00037 55o cinqüenta e cinco
00060 0003c 60o sessenta
00070 00046 70o setenta
00071 00047 71o setenta e um
00075 0004b 75o setenta e cinco
00080 00050 80o oitenta
00081 00051 81o oitenta e um
00085 00055 85o oitenta e cinco
00090 0005a 90o noventa
00091 0005b 91o noventa e um
00095 0005f 95o noventa e cinco
00100 00064 100o cem
00101 00065 101o cento e um
00110 0006e 110o cento e dez
00125 0007d 125o cento e vinte e cinco
00150 00096 150o cento e cinqüenta
00155 0009b 155o cento e cinqüenta e cinco
00170 000aa 170o cento e setenta
00180 000b4 180o cento e oitenta
00190 000be 190o cento e noventa
00200 000c8 200o duzentos
00250 000fa 250o duzentos e cinqüenta
00300 0012c 300o trezentos
00400 00190 400o quatrocentos
00500 001f4 500o quinhentos
00600 00258 600o seiscentos
00700 002bc 700o setecentos
00800 00320 800o oitocentos
00900 00384 900o novecentos
01000 003e8 1000o mil
99999 1869f 99999o noventa nove mil novecentos e noventa e nove 
```
Can you verify the outcomes for me?

If you want to test it yourself, I'll attach [fc-portuges.zip](https://github.com/user-attachments/files/17435919/fc-portuges.zip), which you can place next to your document to verify as well, I assume. PS: had to zip it in order to share

**Comment by ederguidsa** [comment]
> Is that incorrect as well, or is this behavior for beyond the 20s? I can confirm that this behavior is also happening for the 50s:

Yes. Just the 21 is correct. All others must have the "e", as below:
```
00021 00015 21o vinte e um
00022 00016 22o vinte e dois
00023 00017 23o vinte e três
00024 00018 24o vinte e quatro
00025 00019 25o vinte e cinco
```
> Can you verify the outcomes for me?

Yes. All are correct, except the last one. The "e" must appear in the tens of thousands as well, as below:

```
99999 1869f 99999o noventa e nove mil novecentos e noventa e nove 
55555 XXXX 55555o cinqüenta e cinco mil quinhentos e cinqüenta e cinco
```

**Comment by MacLotsen** [comment]
Thanks for the quick response. I've added some extra use-cases ~~and I think there's another issue, namely~~:
```
01000 003e8 1000o mil
01111 00457 1111o mil e cento e onze
11111 02b67 11111o onze mil e cento e onze
55555 0d903 55555o cinquenta e cinco mil e quinhentos e cinquenta e cinco
99999 1869f 99999o noventa e nove mil e novecentos e noventa e nove
```
~~Is `1111` correct like that?~~

**Comment by MacLotsen** [comment]
Will be included in version 3.09
